src/shipit_static_analysis/mozilla_clang.nix

```diff
@@ -0,0 +1,112 @@
+{ releng_pkgs }:
+
+let
+  inherit (releng_pkgs.pkgs) fetchurl gnutar llvmPackages_4;
+  inherit (releng_pkgs.pkgs.stdenv) mkDerivation;
+  inherit (releng_pkgs.pkgs.lib) licenses ;
+
+  clang = llvmPackages_4.clang-unwrapped;
+
+  # Retrieve Mozilla clang-plugin from mozilla central
+  moz_clang_plugin = mkDerivation rec {
+    name = "moz-clang-plugin";
+    revision = "47f7b6c64265";
```

Comment:
So we're pinning our `clang-tidy` version on a fixed mozilla-central revision. Do we plan to update this revision regularly?

It looks like we're either committing recurring engineering time to keeping this up-to-date, or causing our bot to stay with an outdated version of `clang-tidy` if we stop (or forget about) updating it. Not sure if there is a good solution to this.

Reply:
Yeap, we have to pin the mozilla-central revision so that Nix can reliably build the exact same version every time (there is an sha256 download check just below).

As clang-plugin is only updated every few weeks, i think we are fine for now.

Reply:
Thanks for your reply! It would be amazing to have some sort of https://greenkeeper.io service for Nix that can watch dependency revisions and open pull requests for any updates.